register/views.py

```diff
@@ -0,0 +1,28 @@
+from rest_framework.response import Response
+from rest_framework.decorators import api_view, authentication_classes, permission_classes
+from rest_framework import status
+
+from django.db.utils import IntegrityError
+from register.serializers import RegistrationSerializer
+
+@api_view(['POST', ])
+@authentication_classes([])
+@permission_classes([])
+def registration_view(request):
+    serializer = RegistrationSerializer(data=request.data)
+    data = {}
+    if serializer.is_valid():
+        try:
+            user = serializer.save()
+        except IntegrityError:
+            data['response'] = 'User already registered! Please use a different email to register.'
+            return Response(data, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+        data['response'] = 'Successfully registered new user.'
+        data['email'] = user.email
+    else:
+        data = serializer.errors
+    if 'password' in data:
+        return Response(data, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
```suggestion
        data = serializer.errors
        return Response(data, status=status.HTTP_400_BAD_REQUEST)
```

how about langsung 400 aja kalau serializer not valid (?)

Reply:
Aku juga engak paham andrew, tapi is_validate() nya kurang robust.
Meskipun raise errors, dia tetap anggap is_valid gitu.